pkg/providers/cloudstack/cloudstack.go

```diff
@@ -563,12 +563,14 @@ func buildTemplateMapCP(clusterSpec *cluster.Spec, datacenterConfigSpec v1alpha1
 		"cloudstackControlPlaneTemplateOfferingId":   controlPlaneMachineSpec.Template.Id,
 		"cloudstackControlPlaneTemplateOfferingName": controlPlaneMachineSpec.Template.Name,
 		"cloudstackControlPlaneCustomDetails":        controlPlaneMachineSpec.UserCustomDetails,
-		"affinityGroupIds":                           controlPlaneMachineSpec.AffinityGroupIds,
+		"cloudstackControlPlaneAffinity":             controlPlaneMachineSpec.Affinity,
+		"cloudstackControlPlaneAffinityGroupIds":     controlPlaneMachineSpec.AffinityGroupIds,
```

Comment:
i think we need some unit tests to make sure we're actually passing this on to capc

Reply:
+1